java/core/src/test/java/co/worklytics/psoxy/storage/impl/BulkDataSanitizerImplTest.java

```diff
@@ -417,4 +417,54 @@ void pre_v0_4_30_bulk_pseudonym_URL_SAFE_TOKEN_ENCODING(String identifier) {
         assertEquals("t~" + legacyEncoded + (pseudonymizedIdentity.getDomain() == null ? "" : ("@" + pseudonymizedIdentity.getDomain())),
             urlSafeTokenPseudonymEncoder.encode(pseudonymizedIdentity.asPseudonym()));
     }
+
+
+    @SneakyThrows
+    @Test
+    void transform_ghusername() {
+
+        final String EXPECTED = "EMPLOYEE_ID,EMPLOYEE_EMAIL,DEPARTMENT,SNAPSHOT,MANAGER_ID,JOIN_DATE,LEAVE_DATE,GITHUB_USERNAME\r\n" +
+            "2,bob@workltyics.co,Sales,2023-01-06,1,2020-01-01,,\r\n" +
+            "1,alice@worklytics.co,Engineering,2023-01-06,,2019-11-11,,\"{\"\"scope\"\":\"\"github\"\",\"\"hash\"\":\"\"kwv9cWxo7TDgrt1qCegIJv7rA84s_895L_wG_y8hYjA\"\"}\"\r\n" +
+            "4,,Engineering,2023-01-06,1,2018-06-03,,\r\n" +
+            "3,charles@workltycis.co,Engineering,2023-01-06,1,2019-10-06,2022-12-08,\r\n";
+
+        CsvRules rules = CsvRules.builder()
+            .columnsToDuplicate(Map.of("EMPLOYEE_EMAIL", "GITHUB_USERNAME"))
+            .columnsToTransform(Map.of("GITHUB_USERNAME", ColumnarRules.FieldValueTransform.builder()
+                                                                        .filterRegex("(.*)@worklytics.co")
+                                                                        .outputTemplate("%s_enterprise")
+                                                                        .build()))
+            .columnsToPseudonymizeWithScope(Map.of("GITHUB_USERNAME", "github"))
```

Comment:
this is a point for discussion; I've split the definition of the transform we want into essentially 3 steps that must be configured independently, instead of one 'transform' structure encoding those steps. 

advantage of this approach is that it keeps the usual, simple case simple: just put a list of `columnsToPseudonymize` and you're done; and there's no *other* way to pseudonymize that having the column in that list.

biggest con of this approach is that the order is not obvious; you have to know how it's implementation.

alternatives:

```yaml
- columnsToDuplicateAndTransform:
    EMPLOYEE_EMAIL:
         newColumn: GITHUB_USERNAME
         transforms: 
               - filter: "(.*)@worklytics.co"
               - template: "%s_enterprise"
               - pseudonymizeWithScope: 'github'
```
(Actual yaml would be more verbose than that, unless FieldValueTransform is java object with all the properties, with expectation that in practice only use one)

Reply:
I was thinking in order too, 2nd approach is more obvious in that regard, easier to follow.

Reply:
Yeah I think the second one fits better; sometimes order is hard to follow when creating rules.